pure_powervc.py

```diff
@@ -33,9 +33,43 @@
 from powervc_cinder.volume.discovery_driver import PORT_LOCATION
 from powervc_cinder.volume.discovery_driver import PORT_STATUS
 from powervc_cinder.volume.discovery_driver import UNKNOWN_VALUE
+from powervc_cinder.db import api as powervc_db_api
+from oslo_utils import excutils, importutils
+from cinder import exception
+from powervc_cinder import exception as powervc_exception
+from oslo_utils import units
+
+import json
+import six
+
+RESTRICTED_METADATA_VDISK_ID_KEY = "vdisk_id"
+RESTRICTED_METADATA_VDISK_UID_KEY = "vdisk_uid"
+RESTRICTED_METADATA_VDISK_NAME_KEY = "vdisk_name"
 
 LOG = logging.getLogger(__name__)
 
+def create_restricted_metadata(vol_arg=1):
+    """add restricted metadata for a new volume"""
+    def wrap(f):
+        def decorator(*args, **kwargs):
+            args[0]._validate_type(args[vol_arg])
```

Comment:
I don't see a function definition for `_validate_type` so this will cause the driver to crash